### PR TITLE
Feature/fix wagtail npm dependencies

### PR DESCRIPTION
**Pull request by xtine**
## Fix Wagtail JS Dependency Issues
- Addresses https://github.com/18F/fec-cms/issues/1415 and https://github.com/18F/fec-cms/issues/1384
- Fixes wagtail admin HTML editing bug and updates jquery version

## Impacted areas of the application
- Instead of including all `node_modules` in static, we are just including a folder of symlinked npm modules that wagtail admin relies on.
  - Adds `ace-builds`, `rangy`, and `jquery` as wagtail admin npm dependencies.
- Add a wagtail admin override template to include latest jquery (as specified in `package.json`).
  - Moved fec apps above wagtail apps in `INSTALLED_APPS` as fec apps need to declared above wagtail admin app for override to work properly in settings file. Doesn't seem to affect overall application.
- Added a comment in `requirements.txt` so that when wagtail is ever upgraded to note that wagtailadmin override template might need to be updated as well.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+List related PRs against other branches:
+
+branch | PR
+------ | ------
+fix/other_pr | [link]()
+feature/other_pr | [link]()
+
+____
+
+**IF ANY OF THE ABOVE FIELDS DO NOT APPLY, PLEASE DELETE THEM BEFORE SUBMITTING**
```
Comment (ccostino) [inline .github/PULL_REQUEST_TEMPLATE.md]: Nice, thanks for including this!

File: LICENSE.md [hunk 2]
```diff
@@ -1,20 +1,81 @@
...
+- Side navigation
+
+### Contributions to this project
+
+As stated in [CONTRIBUTING](CONTRIBUTING.md), all contributions to this project will be released under the CC0 dedication. By submitting a pull request, you are agreeing to comply with this waiver of copyright interest.
+
+
+
+
+
```
Comment (ccostino) [inline LICENSE.md]: Oh good call on the `LICENSE` update and related documents!  Is this coming from text that we've vetted previously with other projects, or tailored specifically for this one?

File: fec/fec/settings/base.py [hunk 3]
```diff
@@ -198,6 +198,9 @@
 STATICFILES_DIRS = (
     os.path.join(BASE_DIR, 'dist', 'fec', 'static'),
     os.path.join(PROJECT_DIR, 'static'),
+    # We will need just the assets of certain npm modules for Wagtail
+    # So we are including af older of symlinks to modules to access
+    os.path.join(BASE_DIR, 'fec', 'wagtail_npm_dependencies'),
```
Comment (ccostino) [inline fec/fec/settings/base.py]: Nicely done!  I see the symlinks below, nice and simple. :-)

File: fec/fec/templates/wagtailadmin/admin_base.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    <script src="{% static 'wagtailadmin/js/vendor/jquery-ui-1.10.3.js' %}"></script>
+    <script src="{% static 'wagtailadmin/js/vendor/jquery.datetimepicker.js' %}"></script>
+    <script src="{% static 'wagtailadmin/js/vendor/jquery.autosize.js' %}"></script>
+    <script src="{% static 'wagtailadmin/js/vendor/bootstrap-transition.js' %}"></script>
+    <script src="{% static 'wagtailadmin/js/vendor/bootstrap-modal.js' %}"></script>
+    <script src="{% static 'wagtailadmin/js/vendor/bootstrap-tab.js' %}"></script>
+    <script src="{% static 'wagtailadmin/js/vendor/jquery.dlmenu.js' %}"></script>
+    <script src="{% static 'wagtailadmin/js/vendor/tag-it.js' %}"></script>
+    <script src="{% static 'wagtailadmin/js/core.js' %}"></script>
+    {% hook_output 'insert_global_admin_js' %}
```
Comment (ccostino) [inline fec/fec/templates/wagtailadmin/admin_base.html]: What happened to `ace-builds` here?  Is it handled by `{% hook_output 'insert_global_admin_js' %}` now?

File: requirements.txt [hunk 5]
```diff
@@ -10,6 +10,7 @@ psycopg2==2.7.1
 requests==2.13.0
 slacker==0.8.6
 whitenoise==2.0.3
+# before updating wagtail, look at wagtailadmin template override
```
Comment (ccostino) [inline requirements.txt]: Thank you for including this note to remind folks!

File: LICENSE.md (hunk 2 above)
Comment (xtine) [reply LICENSE.md]: This was from @jenniferthibault's PR: https://github.com/18F/fec-cms/pull/1454

File: fec/fec/templates/wagtailadmin/admin_base.html (hunk 4 above)
Comment (xtine) [reply fec/fec/templates/wagtailadmin/admin_base.html]: It's still defined in `fec/fec/wagtail_hooks.py`

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (xtine) [reply .github/PULL_REQUEST_TEMPLATE.md]: More thanks to @jenniferthibault's PR https://github.com/18F/fec-cms/pull/1453 that puts this back in. :)

File: fec/fec/settings/base.py [hunk 6]
```diff
@@ -198,6 +198,9 @@
 STATICFILES_DIRS = (
     os.path.join(BASE_DIR, 'dist', 'fec', 'static'),
     os.path.join(PROJECT_DIR, 'static'),
+    # We will need just the assets of certain npm modules for Wagtail
+    # So we are including a folder of symlinks to modules for access
+    os.path.join(BASE_DIR, 'fec', 'wagtail_npm_dependencies'),
```
Comment (patphongs) [inline fec/fec/settings/base.py]: I may be missing something here, but are the npm dependencies only going to be wagtail from now on? Just wondering why the folder isn't more generic to be `npm_dependencies`

File: fec/fec/templates/wagtailadmin/admin_base.html (hunk 4 above)
Comment (ccostino) [reply fec/fec/templates/wagtailadmin/admin_base.html]: Oh d'oh!  I didn't think to expand the lines in that file to see in the diff, sorry. :-)

File: fec/fec/settings/base.py (hunk 6 above)
Comment (patphongs) [reply fec/fec/settings/base.py]: Got it! Thank you for explaining!

Comment (ccostino) [reply fec/fec/settings/base.py]: You're very welcome! :-)

**Pull request merged**